migrations/20230116124351_alter_identities.up.sql

```diff
@@ -0,0 +1,37 @@
+-- alter identities id and add provider_id column 
+
+do $$
+begin
+    -- rename identities.id to identities.provider_id
+    if exists (select *
+        from information_schema.columns
+        where table_schema = '{{ index .Options "Namespace" }}' 
+        and table_name='identities' and column_name = 'id'
+    ) and not exists (select *
+        from information_schema.columns
+        where table_schema = '{{ index .Options "Namespace" }}' 
+        and table_name='identities' and column_name = 'provider_id'
+    )
+    then
+        alter table "{{ index .Options "Namespace" }}"."identities" rename column "id" to "provider_id";
+    end if;
+
+    alter table {{ index .Options "Namespace" }}.identities
+    add column if not exists id bigint generated by default as identity;
+
+    -- make identities.id a primary key if no primary key constraint is found
+    if not exists ( select constraint_name
+        from information_schema.constraint_column_usage
+        where table_schema = '{{ index .Options "Namespace" }}' and table_name='identities' and constraint_name='identities_pkey'
+    ) then
+        alter table "{{ index .Options "Namespace" }}"."identities" add primary key(id);
+    end if;
+
+    if not exists ( select constraint_name 
+        from information_schema.constraint_column_usage
+        where table_schema = '{{ index .Options "Namespace" }}' and table_name='identities' and constraint_name='provider_provider_id_key'
+    ) then
+        alter table "{{ index .Options "Namespace" }}"."identities"
+        add constraint provider_provider_id_key unique(provider, provider_id);
+    end if;
+end $$;
```

Comment:
Is this really necessary? I'm wary of changing any primary keys after the last email uniqueness problems. Can we just use the existing names of the columns, and only update the Go code to fix readability if that's the goal.

Reply:
decided to remove it since it's not necessary